internal/util/search/search.go

```diff
@@ -0,0 +1,178 @@
+package search
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/GoogleContainerTools/kpt/pkg/kptfile"
+	"sigs.k8s.io/kustomize/kyaml/errors"
+	"sigs.k8s.io/kustomize/kyaml/kio"
+	"sigs.k8s.io/kustomize/kyaml/kio/kioutil"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+const PathDelimiter = "."
+
+// SearchReplace struct holds the input parameters and results for
+// Search and Replace operations on resource configs
+type SearchReplace struct {
+	// Value is the value of the field to be matched
+	Value string
```

Comment:
any good reason we want to drop the By* and Put* prefix in field names here? It's more clear how they relate to the flags and creates logical group.

Reply:
Updated.